pkg/flink/internal/controller/input_controller.go

```diff
@@ -213,3 +178,77 @@ func (c *InputController) toggleSmartCompletion() {
 
 	components.PrintSmartCompletionState(c.getSmartCompletion(), maxCol)
 }
+
+func (c *InputController) getKeyBindings() []prompt.Option {
+	osSpecificBindings := getMacBindings()
+	if runtime.GOOS == "windows" {
+		osSpecificBindings = getWindowsBindings()
+	}
+	return append(
+		[]prompt.Option{
+			prompt.OptionAddKeyBind(prompt.KeyBind{
+				Key: prompt.ControlD,
+				Fn: func(b *prompt.Buffer) {
+					c.shouldExit = true
+				},
+			}),
+			prompt.OptionAddKeyBind(prompt.KeyBind{
+				Key: prompt.ControlQ,
+				Fn: func(b *prompt.Buffer) {
+					c.shouldExit = true
+				},
+			}),
+			prompt.OptionAddKeyBind(prompt.KeyBind{
+				Key: prompt.ControlS,
+				Fn: func(b *prompt.Buffer) {
+					c.toggleSmartCompletion()
+				},
+			}),
+			prompt.OptionAddKeyBind(prompt.KeyBind{
+				Key: prompt.ControlR,
+				Fn: func(b *prompt.Buffer) {
+					c.reverseISearchEnabled = true
+				},
+			}),
+		}, osSpecificBindings...)
+}
+
+func getMacBindings() []prompt.Option {
+	return []prompt.Option{
+		prompt.OptionAddASCIICodeBind(
+			prompt.ASCIICodeBind{
+				ASCIICode: []byte{0x1b, 0x62}, // Alt/Option + Arrow Left
+				Fn:        prompt.GoLeftWord,
+			},
+			prompt.ASCIICodeBind{
+				ASCIICode: []byte{0x1b, 0x66}, // Alt/Option + Arrow Right
+				Fn:        prompt.GoRightWord,
+			},
+			prompt.ASCIICodeBind{
+				ASCIICode: []byte{0x1b, 0x7F}, // Alt/Option + Backspace
+				Fn:        prompt.DeleteWord,
+			},
+		),
+	}
+}
+
+func getWindowsBindings() []prompt.Option {
+	return []prompt.Option{
+		prompt.OptionAddKeyBind(
+			prompt.KeyBind{
+				Key: prompt.ControlLeft,
```

Comment:
Is this the default shortcut on windows for navigating words? I thought alt + arrow was broken and that was what we wanted to fix

Reply:
Nop, ctrl + key is the windows default shortcut. Just tested it too